packages/netlify-cms-backend-github/src/API.js

```diff
@@ -345,33 +355,47 @@ export default class API {
     });
   }
 
-  deleteFile(path, message, options = {}) {
-    const branch = options.branch || this.branch;
+  getFileSha(path, branch) {
+    /**
+     * We need to request the tree first to get the SHA. We use extended SHA-1
+     * syntax (<rev>:<path>) to get a blob from a tree without having to recurse
+     * through the tree.
+     */
+
     const pathArray = path.split('/');
     const filename = last(pathArray);
     const directory = initial(pathArray).join('/');
     const fileDataPath = encodeURIComponent(directory);
     const fileDataURL = `${this.repoURL}/git/trees/${branch}:${fileDataPath}`;
-    const fileURL = `${this.repoURL}/contents/${path}`;
 
-    /**
-     * We need to request the tree first to get the SHA. We use extended SHA-1
-     * syntax (<rev>:<path>) to get a blob from a tree without having to recurse
-     * through the tree.
-     */
     return this.request(fileDataURL, { cache: 'no-store' }).then(resp => {
       const { sha } = resp.tree.find(file => file.path === filename);
+      return sha;
+    });
+  }
+
+  deleteFile(path, message, options = {}) {
+    const branch = options.branch || this.branch;
+
+    return this.getFileSha(path, branch).then(sha => {
       const opts = { method: 'DELETE', params: { sha, message, branch } };
       if (this.commitAuthor) {
         opts.params.author = {
           ...this.commitAuthor,
           date: new Date().toISOString(),
         };
       }
+      const fileURL = `${this.repoURL}/contents/${path}`;
       return this.request(fileURL, opts);
     });
   }
 
+  createBranchAndPullRequest(branchName, sha, commitMessage) {
```

Comment:
Overridden in `GraphQLAPI` and replaced by one mutation requests instead of two REST API requests